test/core/end2end/fixtures/http_proxy_fixture.c

```diff
@@ -352,6 +394,30 @@ static void on_read_request_done(grpc_exec_ctx* exec_ctx, void* arg,
     GRPC_ERROR_UNREF(error);
     return;
   }
+  // If proxy auth is being used, check if the header is present and as expected
+  if(grpc_channel_args_find(
+      conn->proxy->channel_args,
+      GRPC_END2END_HTTP_PROXY_TEST_CONNECT_AUTH_PRESENT) != NULL) {
+    bool auth_header_found = false;
```

Comment:
There's probably no need for this bool.  Instead, just move the body of the `if (!auth_header_found)` block inside of the loop:

```
if (strcmp(conn->http_request.hdrs[i].key, "Proxy-Authorization") == 0) {
  if (!proxy_auth_header_matches(
          exec_ctx, conn->http_request.hdrs[i].value)) {
    // ... call proxy_connection_failed()...
    return;
  }
  break;
}
```